packages/server-admin-ui/src/views/ServerConfig/BasicProvider.js

```diff
@@ -316,6 +316,33 @@ class RemoveNullsInput extends Component {
   }
 }
 
+class AppendChecksum extends Component {
+  render() {
+    return (
+      <FormGroup row>
+        <Col xs='3' md='2'>
+          <Label>Append Checksum</Label>
+        </Col>
+        <Col xs='2' md='3'>
+          <Label className='switch switch-text switch-primary'>
+            <Input
+              type='checkbox'
+              name='options.appendChecksum'
+              helpText='Append checksum if missing'
+              className='switch-input'
+              onChange={event => this.props.onChange(event)}
+              checked={this.props.value.appendChecksum && !this.props.value.validateChecksum}
```

Comment:
The checked status must reflect only the `appendChecksum` value, there's a bug here. Try this sequence
- turn validate checksum off
- turn append checksum on
- turn validate checksum on => append checksum is shown as off but **not actually changed**
- Apply to save the settings => both validate and append are `true` in settings!

My suggestion about making the checkbox disabled isn't helping much, as the button in off state looks exactly like it does in disabled state. What if we add a help text to both, saying 'Turn validate checksum off to enable appending the checksum' and 'Turn append checksum off to enable checksum validation'? This would solve the above mentioned problem.